src/macros.jl

```diff
@@ -0,0 +1,56 @@
+for struct_type in (:Struct, :Mutable, :CustomStruct, :OrderedStruct, :AbstractType, :DictType, :ArrayType, :StringType, :NumberType, :BoolType, :NullType)
+    @eval begin
+        @doc """
+            @$($struct_type)(expr::Expr)
```

Comment:
From the tests, it seems like we're over-interpolating here; maybe change to `@$(Meta.quot($struct_type))`

Reply:
I think it's possibly to do with the macros being evaluated before the struct types exist - moving the `include` statement to the end of `StructTypes.jl` seems to work. 

`@$(Meta.quot($struct_type))` made the docs look a bit funny:
```
help?> StructTypes.@DictType
  @:(StructTypes.DictType)(expr::Expr)
```

Reply:
ah, that makes sense. do you want to push your change then?